mix.exs

```diff
@@ -22,7 +31,7 @@ defmodule OpentelemetryAbsinthe.MixProject do
 
   defp deps do
     [
-      {:absinthe, ">= 1.5.0", optional: true},
+      {:absinthe, "~> 1.7.0", optional: true},
```

Comment:
I assume this won't lock clients from using older versions (`optional` always confuses me for some reason)

Also, what about moving in to `dev_deps`?

Reply:
> I assume this won't lock clients from using older versions

it will. The version of otel_absinthe on this branch only works with absinthe 1.7.0, not 1.5.0, since absinthe doesn't provide a stable api for telemetry 

If we really want to support multiple versions of absinthe we would need to have different code to support different versions, figure out how to matrix test every absinthe version and it just doesn't seem to me like that's worth the additional maintenance burden

Reply:
We don't want to move this to dev_deps, since while this library technically doesn't require absinthe, it will not work with an unsupported version, so we need to actually have mix verify this

Reply:
So my doubts were founded! I think it's fine, as you say it's not worth the burden since it's not like we're bumping a major